app/models/search.rb

```diff
@@ -25,9 +25,20 @@ def highlights
     end.to_h
   end
 
+  ##
+  # Returns a modified result list from Solr, by following the following:
+  # - Take the last 5 results
+  # - Remove duplicates by case (e.g. The, the)
+  # - Sort by "weight" (occurance) and then by length. This helps to sort values
+  #   for recurring sequences.
+  #   "the cou": ["the Court", "the Court decided", "the Court decided to"]
+  # rubocop:disable Metrics/AbcSize
   def suggestions
-    suggest_response['suggest'].values.first[q]['suggestions']
+    suggest_response['suggest'].values.first[q]['suggestions'].take(5)
```

Comment:
Hmm, interesting.  What is the reason we're not sorting/de-duping before we `.take(5)`? I would imagine sorting/de-duping before hand would ensure that we get the best 5 results (as opposed to getting the sorted/de-duped first 5), but I haven't looked too closely at the suggestion response.

Reply:
Yeah I hear you. The results are already coming back sorted by weight, so hopefully we are "good" on that regard. The reason I chose to take before hand was to hopefully improve performance here. Instead of having resort and downcase on 400 results, we now only have to do so on 5. This does mean that we may lose a result or two. In testing this I am seeing that "The" "the" are removed so that sometimes only 4 autocomplete results are shown. Maybe my performance optimization was premature, but that was my reasoning.

Reply:
Hm. Deduplicating  by case /should/ be happening in our solr analyzer already, I thought. Maybe we can take a look sometime today.

Reply:
Yep lets look at it, but I don't think it is happening.

Reply:
I wonder if we also wantto deduplicate suggestions that produce no more hits than a slightly longer phrase (e.g. no reason to suggest both "Contrary to" and "Contrary to the"). That may suggest doing the sorting/deduping pass earlier anyway.

Reply:
Well.. maybe it is.

Reply:
@jkeck updated this to take at the end now